torch/_dynamo/utils.py

```diff
@@ -219,185 +207,111 @@ def print_time_report() -> None:
     print(out)
 
 
-def _add_time_spent(key: str, phase_name: str, time_spent: float) -> None:
-    frame_phase_timing[key][phase_name] += time_spent
-
-
-# Use frame_phase_timing to record remote_cache_time_saved
-# This follows the same principles of key as the other frame phase timings,
-# but is incremented by FxGraphCache (and later AOTAutogradCache) directly
-def add_remote_cache_time_saved(time_saved_ns: int, is_backward: bool = False) -> None:
-    key = None
-    if is_backward:
-        # Use compile id as the frame key for backwards compilation
-        key = str(torch._guards.CompileContext.current_compile_id())
-    else:
-        key = str(curr_frame)
-    # Convert to seconds (as a float)
-    time_saved = time_saved_ns / 1e9
-    _add_time_spent(key, "remote_cache_time_saved", time_saved)
+# Use the following singleton to capture and log CompilationMetrics. Entering the context
+# manager allocates a new record to be logged when it exits. (You should not need to use
```

Comment:
claim: instead, `dynamo_timed` should be preferentially used to cause new records to be allocated, no reason to use this singleton directly.

Reply:
I think the difficulty for me is there's currently a sensible spot where we set all the fields that should be set "globally" once for the log entry. IMO, it's not as clear if dynamo_timed is quietly creating the context for you to store those in.

Reply:
I think in my ideal terminal state, most of the fields are being set (1) in `dynamo_timed`, so they obviously are always set, or (2) done via aggregation, so they're automatically picked up. OK to leave as is for now.